streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamReduce.java

```diff
@@ -88,14 +95,13 @@ public void process(final K key, final V value) {
             }
 
             // update the store with the new value
-            store.put(key, newAgg);
+            store.put(key, ValueAndTimestamp.make(newAgg, context().timestamp()));
```

Comment:
Today we store the timestamp in cache entry and upon flushing we call `entry.entry().context().timestamp()` to set the timestamp. Would that be replaced with the underlying store's timestamp then?

Reply:
While working on a follow PR, I figure out the details. The `TimestampedForwardingCacheFlushListener` needs to extract the timestamp from the value, and set it as record timestamp. Similar to `SessionCacheFlushListener` in another PR: https://github.com/apache/kafka/pull/6645/files#diff-4039efe54e3b87d65c20bd7d14c1b149R43